### PR TITLE
Refine pai training directly read maxcompute table

### PR DESCRIPTION
**Pull request by typhoonzero**
Can run pai job directly read a maxcompute table.

### CODE REVIEW

File: pkg/sql/codegen/pai/codegen.go [hunk 1]
```diff
@@ -18,34 +18,76 @@ import (
...
 	"sqlflow.org/sqlflow/pkg/sql/codegen/tensorflow"
 	"sqlflow.org/sqlflow/pkg/sql/ir"
 )
 
 const entryFile = "entry.py"
 
+func getTableFromSelect(dataSource, trainSelect string) (string, string, error) {
+	// FIXME(typhoonzero): copied from tensorflow/codegen.go, should remove this and use the temp table
+	// in the workflow
+	fromRegex, err := regexp.Compile("FROM[\\s\\n]+([\\w\\.]*)")
```
Comment (Yancey0623) [inline pkg/sql/codegen/pai/codegen.go:36]: Maybe we need a strict check to make sure the `trainSelect` should be `SELECT * FROM tmp_table TO TRAIN...`, the standard sql statement should not contain any `WHERE/LIMIT/...`.

File: python/sqlflow_submitter/tensorflow/train.py [hunk 2]
```diff
@@ -201,6 +224,14 @@ def validate_input_fn(batch_size):
                 print("%s: %s" % (k, v[-1]))
         classifier.save_weights(save, save_format="h5")
     else:
+        # if is_pai:
```
Comment (Yancey0623) [inline python/sqlflow_submitter/tensorflow/train.py]: Is this a TODO comment?

File: pkg/sql/codegen/pai/codegen.go [hunk 3]
```diff
@@ -18,34 +18,76 @@ import (
...
+	if err != nil {
+		return "", "", err
+	}
+	matches := fromRegex.FindAllStringSubmatch(trainSelect, -1)
+	if len(matches) != 1 {
+		return "", "", fmt.Errorf("only support simple SQL query, but got %s", trainSelect)
+	}
+	paiTableFull := matches[0][1]
+	paiDatabase := ""
+	paiTable := ""
```
Comment (shendiaomo) [inline pkg/sql/codegen/pai/codegen.go]: How about renaming these variables with an `odps` prefix or without any prefix? After all, we are in the `pai` package,a `pai` prefix is redundant.

File: pkg/sql/codegen/pai/codegen.go [hunk 4]
```diff
@@ -18,34 +18,76 @@ import (
...
 	var tpl = template.Must(template.New("Submit").Parse(tfWrapperTmplText))
 	filler := wrapperFiller{
-		DataSource: dataSource,
-		ModelName:  modelName,
-		EntryFile:  entryFile,
+		DataSource:  dataSource,
+		ModelName:   modelName,
+		EntryFile:   entryFile,
+		PAIDatabase: paiDatabase,
+		PAITable:    paiTable,
```
Comment (shendiaomo) [inline pkg/sql/codegen/pai/codegen.go]: ditto.

File: python/sqlflow_submitter/tensorflow/train.py [hunk 5]
```diff
@@ -148,22 +145,39 @@ def input_fn(datasetStr):
...
-        driver, dsn = datasource.split("://")
-        _, _, _, database = parseMaxComputeDSN(dsn)
-        tables = ["odps://%s/tables/%s" % (database, pai_table)]
+        table_parts = pai_table.split(".")
+        if len(table_parts) == 2:
+            database, table_name = table_parts
+        elif len(table_parts) == 1:
+            table_name = pai_table
+            driver, dsn = datasource.split("://")
+            _, _, _, database = parseMaxComputeDSN(dsn)
```
Comment (shendiaomo) [inline python/sqlflow_submitter/tensorflow/train.py]: IMO `database = parseMaxComputeDSN(dsn)[-1]` is more pythonic.

File: pkg/sql/codegen/pai/codegen.go (hunk 1 above)
Comment (typhoonzero) [reply pkg/sql/codegen/pai/codegen.go:36]: Will fix this in next PR

File: python/sqlflow_submitter/tensorflow/train.py (hunk 2 above)
Comment (typhoonzero) [reply python/sqlflow_submitter/tensorflow/train.py]: removed.

**Review by shendiaomo — APPROVED** [review]

**Pull request merged**